test/micro/org/openjdk/bench/java/math/FpRoundingBenchmark.java

```diff
@@ -26,53 +26,91 @@
 import org.openjdk.jmh.annotations.*;
-import org.openjdk.jmh.infra.Blackhole;
 
 @OutputTimeUnit(TimeUnit.MILLISECONDS)
 @State(Scope.Thread)
 public class FpRoundingBenchmark {
 
-  @Param({"1024"})
+  @Param({"1024", "2048"})
   public int TESTSIZE;
 
   public double[] DargV1;
-
-  public double[] Res;
+  public double[] ResD;
+  public float[] FargV1;
+  public float[] ResF;
 
   public final double[] DspecialVals = {
-      0.0, -0.0, Double.NaN, Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY};
+      0.0, -0.0, Double.NaN, Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY,
+      Double.MAX_VALUE, -Double.MAX_VALUE, Double.MIN_VALUE, -Double.MIN_VALUE,
+      Double.MIN_NORMAL
+  };
+
+  public final float[] FspecialVals = {
+      0.0f, -0.0f, Float.NaN, Float.NEGATIVE_INFINITY, Float.POSITIVE_INFINITY,
+      Float.MAX_VALUE, -Float.MAX_VALUE, Float.MIN_VALUE, -Float.MIN_VALUE,
+      Float.MIN_NORMAL
+  };
 
   @Setup(Level.Trial)
   public void BmSetup() {
-    int i = 0;
-    Random r = new Random(1024);
-    DargV1 = new double[TESTSIZE];
-    Res = new double[TESTSIZE];
-
-    for (; i < DspecialVals.length; i++) {
-      DargV1[i] = DspecialVals[i];
-    }
-
-    for (; i < TESTSIZE; i++) {
-      DargV1[i] = r.nextDouble()*TESTSIZE;
-    }
+      int i = 0;
+      Random r = new Random(1024);
+
+      DargV1 = new double[TESTSIZE];
+      ResD = new double[TESTSIZE];
+
+      for (; i < DspecialVals.length; i++) {
+          DargV1[i] = DspecialVals[i];
+      }
+
+      for (; i < TESTSIZE; i++) {
+          DargV1[i] = r.nextDouble()*TESTSIZE;
+      }
+
+      FargV1 = new float[TESTSIZE];
+      ResF = new float[TESTSIZE];
+
+      for (; i < FspecialVals.length; i++) {
+          FargV1[i] = FspecialVals[i];
+      }
+
+      for (; i < TESTSIZE; i++) {
+          FargV1[i] = r.nextFloat()*TESTSIZE;
+      }
+  }
+
+  @Benchmark
+  public void test_ceil() {
+      for (int i = 0; i < TESTSIZE; i++) {
+          ResD[i] = Math.ceil(DargV1[i]);
+      }
+  }
+
+  @Benchmark
+  public void test_floor() {
+      for (int i = 0; i < TESTSIZE; i++) {
+          ResD[i] = Math.floor(DargV1[i]);
+      }
   }
 
   @Benchmark
-  public void testceil(Blackhole bh) {
-    for (int i = 0; i < TESTSIZE; i++)
-      Res[i] = Math.ceil(DargV1[i]);
+  public void test_rint() {
+      for (int i = 0; i < TESTSIZE; i++) {
+          ResD[i] = Math.rint(DargV1[i]);
+      }
   }
 
   @Benchmark
-  public void testfloor(Blackhole bh) {
-    for (int i = 0; i < TESTSIZE; i++)
-      Res[i] = Math.floor(DargV1[i]);
+  public void test_round_double() {
+      for (int i = 0; i < TESTSIZE; i++) {
+          ResD[i] = Math.round(DargV1[i]);
+      }
   }
 
   @Benchmark
-  public void testrint(Blackhole bh) {
-    for (int i = 0; i < TESTSIZE; i++)
-      Res[i] = Math.rint(DargV1[i]);
+  public void test_round_float() {
+      for (int i = 0; i < TESTSIZE; i++) {
+          ResF[i] = Math.round(FargV1[i]);
+      }
```

Comment:
I think that this is wrong: you should not be storing the result into a float array because it requires an extra integer->float conversion, which distorts the timings. You need `resI` and `resL` for the results of `Math.round`.